src/ZfcUser/Controller/UserController.php

```diff
@@ -348,7 +354,7 @@ public function changeEmailAction()
     public function getUserService()
     {
         if (!$this->userService) {
-            $this->userService = $this->getServiceLocator()->get('zfcuser_user_service');
+            $this->userService = $this->serviceLocator->get('zfcuser_user_service');
```

Comment:
I would argue instead to inject this via the constructor.

If the reason for *not* injecting it is due to it being used only in some execution paths, wrap it as a lazy service.